src/bundle.c

```diff
@@ -727,7 +729,11 @@ int add_subscriptions(struct list *bundles, struct list **subs, struct manifest
 			}
 		} else {
 			// Just add it to a list if it doesn't exist
-			create_and_append_subscription(subs, bundle);
+			if (recursion > 0) {
```

Comment:
This isn't going to work.

If BundleA and BundleB aren't installed and bundleA includes bundleB and you run

    swupd bundle-add bundleB bundleA

we are going to try to install bundleA first and add bundleB as a dependency. And then we try to install bundleB, but this is already listed as a dependency.
At the end we print:

    Successfully installed 2 bundles
    1 bundles was installed as dependency

What's is incorrect.

Instead of that, can't we just report the total number of installed bundles at the end?

Successfully installed X bundles, where X is the number of total bundles installed and not just the ones that were explicitly requested? I wonder if we really need this info.

Reply:
I think even the request is to list deps for a single bundle. I could see a way where the user could tally the expected includes and then run the bundle-add to see she/he gets it. But yes, this is super adhoc request.

The user could also infer the same from total installed bundles with what otavio suggested. Also an added incentive in this case is a much cleaner general solution.

Reply:
I wrote this implementation based on my interpretation of this comment from Matthew "I agree that it would be nice to additionally report number of dependencies installed as a separate metric". 

We could just report the total number of bundles installed at the end as @otaviobp suggested, it would be very easy to implement it that way actually, but I'm not convinced it is the best way to do it.

We are very familiar with how swupd works so if we run:
`swupd bundle-add bundleA`

and after running it we see
`Successfully installed 5 bundles`

we could easily assume swupd installed 4 bundles as dependencies. But I'm not sure another less experienced user would make this assumption, maybe they would think something went wrong.

To make this more difficult... what if you add let's say 3 bundles, and each installs 5 dependencies (some of them could be common among them), then if one of the bundles fail to install you would still see something like:
```
swupd bundle-add bundleA bundleB bundleC
Failed to install 1 of 15 bundles
```

Reply:
@castulo I agree then. Let's go with the solution that prints the dependencies separated from the bundles the user added. just fix the reported problem and it would be nice to have a test covering that too.

Reply:
I'll work on that @otaviobp